packages/dds/tree/src/feature-libraries/sequence-field/compose.ts

```diff
@@ -620,8 +629,10 @@ export class ComposeQueue<T> {
 		for (const mark of newMarks) {
 			if (isObjMark(mark) && mark.type === "Insert") {
 				const newRev = mark.revision ?? this.newRevision;
-				if (newRev !== undefined && deletes.has(newRev)) {
-					this.cancelledInserts.add(newRev);
+				const newIntention = getIntention(newRev, revisionMetadata);
+				if (newIntention !== undefined && deletes.has(newIntention)) {
```

Comment:
When adding to `deletes` shouldn't we be using the intention instead of the revision?